### ISSUE TITLE
GitHub Deployment notifications filter

### ISSUE THREAD

**Issue by derwasp** [integrations/slack#942]
Currently it's possible to create filtered subscription for a bunch of events, but it's not possible for the deployment events.
Since we are deploying quite often, we have way too many notifications and in reality are only looking at the failed ones.

It would be extremely cool if we could filter based on the deployment [state](https://developer.github.com/v3/repos/deployments/#parameters-2), like so:
`/github subscribe myawesomeorg/MyIncredibleProject deployments +status:failure`

**Comment by simonbrandhof** [comment]
Filtering by environment is another use-case for this improvement.

**Comment by arielkirkwood** [comment]
I think it’s still relevant.

**Comment by raghera** [comment]
I think it's still relevant.  We only care about deployments to some envs not all and particularly failures.

**Comment by almed4** [comment]
Agreed! We don't want to spam our whole org with dev deployments, but it's important for API consumers to know when we do releases :)

**Comment by fbjork** [comment]
+1

**Comment by seanconrad1** [comment]
+1

**Comment by earthpyy** [comment]
+1 for this!

**Comment by tughril** [comment]
+1 for this!

**Comment by pratikyadav** [comment]
+1 for Filtering by environment and state

**Comment by kirqz23** [comment]
+1 filtering by env and state

**Comment by jonathan-dunn-telus** [comment]
+1 !!

**Comment by seanconrad1** [comment]
+1

**Comment by jrea** [comment]
+1

**Comment by Julian-Cloudshelf** [comment]
+2

**Comment by tubbo** [comment]
We use preview deployments for our staging environments, so using `/github subscribe deployments` is really noisy for us. Ideally, we'd like to filter deployment notifications to only the `main` branch, so one can see what has been pushed to production.

**Comment by dlazares** [comment]
+1, any updates on this as a feature?

**Comment by styd** [comment]
+1

**Comment by jamie-unmind** [comment]
+1

**Comment by dstoyanoff** [comment]
+1 for env filter

**Comment by berryd** [comment]
+1

**Comment by benjaminpaige** [comment]
+1 pretty please

**Comment by am-cornelis** [comment]
We'd also be using a feature like this, in particular filtering out on the workflow name

**Comment by tihuan** [comment]
+1 still hoping this get bumped in priority, or if there's any pointers on how to implement this, my team might be able to take a look. Thanks!

**Comment by itaispiegel** [comment]
+1

**Comment by jackblanc** [comment]
+1

**Comment by jonathan-dunn-telus** [comment]
+1 
would be really helpful ! Thanks

**Comment by Ibrahim-elouard** [comment]
+1

**Comment by metarsit-coinhako** [comment]
+1

**Comment by vbalazs** [comment]
Please use the reactions to vote up the issue and avoid the unnecessary +1 comments 🙏
They notify all the issue subscribers every time. I always get excited that there is an update about this, just to see that it was an another spammy comment 😿

I don't wish to unsubscribe to this issue since I'm still interested in it.

Since the requirements and popularity here are quite clear, maybe the repo maintainers could consider locking this issue for external comments (and give a much needed update 🙏)

**Comment by igor9silva** [comment]
4 years in and not an option 🥲

*My use case is very simple:* I only want to notify for our production environment.
Isn't it most people use case? What am I missing?

I tried with `workflows` instead of `deployment`, but no option as well.

**Comment by styd** [comment]
Hey, @igor9silva ! We had almost the same need as you. Only we separate the notifications to different channels in slack, e.g. production to `#production-deployment` channel and staging to `#staging-deployment`. We decided to switch to using [this action](https://github.com/slackapi/slack-github-action) in our workflows instead.

**Comment by igor9silva** [comment]
> Hey, @igor9silva ! We had almost the same need as you. Only we separate the notifications to different channels in slack, e.g. production to `#production-deployment` channel and staging to `#staging-deployment`. We decided to switch to using [this action](https://github.com/slackapi/slack-github-action) in our workflows instead.

Thanks for the reference!

**Comment by Harim-T** [comment]
+1

**Comment by landsman** [comment]
what the hell there is no filter for the environment of deployment?

**Comment by opp-svega** [comment]
+1

**Comment by 007vasy** [comment]
+1

**Comment by RZandvoort-VSM** [comment]
+1

**Comment by larsjoakimgrahn** [comment]
Please do not continue commenting with +1 on this issue. Instead use reactions on the initial post. Writing +1 triggers a notification for every person who is following this issue.

**Comment by eliasmistler** [comment]
Are there any updates on this? This really seems like a basic feature that lots of people need, and I'm quite surprised nothing has happened in 4 years :/

**Comment by igor9silva** [comment]
They're to worried with Copilot

**Comment by eliasmistler** [comment]
Actually, looking at the docs, there is a new way to do this:

Don't use the `deployments` notifications:

```
/github unsubscribe my-org/my-repo deployments
```

Instead, use the `worfklows` feature:

```
/github subscribe my-org/my-repo workflows:{name:“deploy_to_prod”}
```

That being said, I couldn't get it to work yet -- if someone does get it to work, please share!

**Comment by psilore** [comment]
I got the subscription workflow to work, but in tandem with releases subscription

Kinda like this,

```
/github subscribe organisation/repository releases, workflows:{ name:"Production" event:"release" }
```

If I understand this correctly, this works because I have a workflow name: Production and it triggers on the event `release`

Example (removed org in screenshot):
![readacted_org-screenshot](https://github.com/integrations/slack/assets/366256/3c88ca48-3f89-495f-9c7a-a803a070b4ef)

However, I dont like that theres a button "Re-run all jobs" because if I send that to a slack channel just to inform, members in that channel can unintentionally and intentionally trigger workflow =( Dont know how to bypass that or remove button
